### PR TITLE
[EBPF] Add extra information to test-json-review output

### PR DESCRIPTION
**Pull request by gjulianm**
### What does this PR do?

Adds owner information to the output of `test-json-review` based on `CODEOWNERS` (only looking at teams, similar to what the CI Visibility product does), and also sorts the list of failed tests alphabetically.

### Motivation

Facilitate parsing test failures in errored jobs and notifying the approriate teams.

### Additional Notes

Job with example failures: https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/595066844

### CODE REVIEW

**Review by FlorentClarret — APPROVED** [review]

**Review by pducolin — APPROVED** [review]

**Comment by usamasaqib** [comment]
I think the approach as-is may create noise for the wrong teams. For example in [this job](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/572068715) tests are incorrectly attributed to windows-kernel-integrations, whereas this team has nothing to do with the test in question.

**Comment by gjulianm** [comment]
> I think the approach as-is may create noise for the wrong teams. For example in [this job](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/572068715) tests are incorrectly attributed to windows-kernel-integrations, whereas this team has nothing to do with the test in question.

You're right, I didn't notice that one. Let me fix that case.

File: test/new-e2e/system-probe/test-json-review/codeowners.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("open %s: %s", path, err)
+	}
+	defer f.Close()
+
+	return loadCodeownersWithReader(f), nil
+}
+
+func loadCodeownersWithReader(contents io.Reader) *codeowners {
```
Comment (brycekahle) [inline test/new-e2e/system-probe/test-json-review/codeowners.go:34]: Is there not a package that supports parsing codeowners?

Comment (gjulianm) [reply test/new-e2e/system-probe/test-json-review/codeowners.go:34]: There is https://github.com/hmarr/codeowners but I thought that given our simplified needs for this, it was better to not bring an external dependency into this.

File: test/new-e2e/go.mod [hunk 2]
```diff
@@ -234,7 +234,7 @@ require (
 	go.opentelemetry.io/otel/trace v1.28.0 // indirect
 	go.starlark.net v0.0.0-20230525235612-a134d8f9ddca // indirect
 	go.uber.org/atomic v1.11.0 // indirect
-	golang.org/x/exp v0.0.0-20240719175910-8a7402abbf56 // indirect
+	golang.org/x/exp v0.0.0-20240719175910-8a7402abbf56
```
Comment (brycekahle) [inline test/new-e2e/go.mod:237]: Can you run `go mod tidy` in this directory? Normally the `indirect` requires are in a separate block.

File: test/new-e2e/system-probe/test-json-review/testowners.go [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+	}
+
+	testName := strings.Split(ev.Test, "/")[0]
+
+	var testFunc *gosym.Func
+	// Test name to function name is not a 1:1 mapping, so we need to iterate over all functions to find the right one
+	for i := range symTable.Funcs {
+		f := &symTable.Funcs[i]
+		// First, check if the function name contains the package name
+		if !strings.Contains(f.Name, ev.Package) {
```
Comment (brycekahle) [inline test/new-e2e/system-probe/test-json-review/testowners.go:110]: Is `Contains` the right operation here, or should it be a prefix?

File: test/new-e2e/system-probe/test-json-review/testowners_test.go [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+	"bytes"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+const ownersfile = `
+# A comment
+/pkg/network @networks
```
Comment (brycekahle) [inline test/new-e2e/system-probe/test-json-review/testowners_test.go:20]: Do you know what happens when there are multiple owners listed? I know we have that a lot.

File: test/new-e2e/go.mod (hunk 2 above)
Comment (gjulianm) [reply test/new-e2e/go.mod:237]: Ran it twice, it leaves it there for some reason.

File: test/new-e2e/system-probe/test-json-review/testowners.go (hunk 3 above)
Comment (gjulianm) [reply test/new-e2e/system-probe/test-json-review/testowners.go:110]: The function is prefixed with the full package name so the package itself (e.g. `pkg/network`) is in the middle of the string. An option would be to remove the prefix `github.com/DataDog/datadog-agent/...` but I think this approach is simpler and works for the package layout we have.

File: test/new-e2e/system-probe/test-json-review/testowners_test.go (hunk 4 above)
Comment (gjulianm) [reply test/new-e2e/system-probe/test-json-review/testowners_test.go:20]: It takes all into account, added a test just in case

**Review by brycekahle — APPROVED** [review]

**Pull request merged**